src/ui/bibliography-renderer/index.ts

```diff
@@ -0,0 +1,39 @@
+import joplin from "api";
+import { ContentScriptType } from "api/types";
+import { DataStore } from "../../data/data-store";
+import { fromRefToAPA } from "../../util/apa.util";
+import { REFERENCE_LIST_CONTENT_SCRIPT_ID } from "../../constants";
+
+/**
+ * Render the full list of references at the end of the note viewer
+ */
+export async function registerBibliographyRenderer (): Promise<void> {
+    
+    /* Register a new content script of type "markdown-it" */
+    await joplin.contentScripts.register(
+        ContentScriptType.MarkdownItPlugin,
+        REFERENCE_LIST_CONTENT_SCRIPT_ID,
+        './ui/bibliography-renderer/render-list-content-script.js'
+    );
+
+    /**
+     * 1- Lookup reference objects based on their IDs (using DataStore)
+     * 2- Convert reference objects into APA format (in HTML)
+     * If some reference objects are not found, ignore them
+     * No need to html-escape the APA output, the library handles that
+     */
+    await joplin.contentScripts.onMessage(REFERENCE_LIST_CONTENT_SCRIPT_ID, (IDs: string[]) => {
+        let refs: string[] = [];
+        IDs = [...new Set(IDs)];
```

Comment:
You're often converting arrays to set and the other way around. Is there any reason for that? Isn't this code pretty much the same as `IDS = IDS`?

Reply:
> Isn't this code pretty much the same as `IDS = IDS`?

Not exactly. Previously, if there are multiple occurrences of the same reference, it will be rendered twice at the bottom.

Reply:
Ok I see. It's not very obvious that it's doing this so maybe add a comment?